File: brawl-pong-server/src/main/kotlin/models/entities/Paddle.kt
package io.sourceempire.brawlpong.models.entities

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonManagedReference
import io.vertx.ext.web.handler.sockjs.SockJSSocket
import java.util.UUID

class Paddle(val x: Float,
             var y: Float,
             val speed: Float,
             val height: Int = 100,
             val width: Int = 20) {
    enum class Direction {
        Stop,
        Up,
        Down
    }

    @JsonIgnore
    var upKeyPressed: Boolean = false

    @JsonIgnore
    var downKeyPressed: Boolean = false

    val direction: Direction
        get() = when {
            upKeyPressed && !downKeyPressed -> Direction.Up
            !upKeyPressed && downKeyPressed -> Direction.Down
            else -> Direction.Stop
        }
}

File: brawl-pong-client/src/types/Paddle.ts
export enum PlayerDirection {
    Stop = "Stop",
    Up = "Up",
    Down = "Down"
}

export type Paddle = {
    x: number,
    y: number,
    direction: PlayerDirection,
    height: number,
    width: number,
    speed: number,
}

export type PlayerInfo = {
    id: string
    ready: boolean,
    connected: boolean,
    isSessionPlayer: boolean,
}

